src/vm-repair/azext_vm_repair/custom.py

```diff
@@ -0,0 +1,252 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from datetime import datetime
+
+from knack.log import get_logger
+
+from azure.cli.command_modules.vm.custom import get_vm, _is_linux_os
+from azure.cli.command_modules.storage.storage_url_helpers import StorageResourceIdentifier
+from msrestazure.tools import parse_resource_id
+
+from .repair_utils import _uses_managed_disk, _call_az_command, _clean_up_resources, _fetch_compatible_sku, _list_resource_ids_in_rg, _get_rescue_resource_tag
+from .exceptions import AzCommandError, SkuNotAvailableError, UnmanagedDiskCopyError
+
+# pylint: disable=line-too-long, too-many-locals, too-many-statements, trailing-whitespace
+
+logger = get_logger(__name__)
+
+def swap_disk(cmd, vm_name, resource_group_name, rescue_password=None, rescue_username=None):
+    # begin progress reporting for long running operation
+    cmd.cli_ctx.get_progress_controller().begin()
+    cmd.cli_ctx.get_progress_controller().add(message='Running')
+
+    target_vm = get_vm(cmd, resource_group_name, vm_name)
+    is_linux = _is_linux_os(target_vm)
+    target_disk_name = target_vm.storage_profile.os_disk.name
+    is_managed = _uses_managed_disk(target_vm)
+
+    if is_linux:
+        os_image_name = "UbuntuLTS"
+    else:
+        # temp fix to mitigate Windows disk signature collision error
+        if target_vm.storage_profile.image_reference \
+           and target_vm.storage_profile.image_reference.version == '2016.127.20190214':
+            os_image_name = "MicrosoftWindowsServer:WindowsServer:2016-Datacenter:2016.127.20190115"
+        else:
+            os_image_name = "MicrosoftWindowsServer:WindowsServer:2016-Datacenter:2016.127.20190214"
+
+    timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+    copied_os_disk_name = vm_name + '-DiskCopy-' + timestamp
+    copied_disk_uri = None
+    rescue_vm_name = ('rescue-' + vm_name)[:15]
+    rescue_rg_name = 'rescue-' + vm_name + '-' + timestamp
```

Comment:
Creating a non-deterministic resource group name will force users to supply rescue VM by ID. Crafting a deterministic resource group would avoid this.

Reply:
I made the resource group non-deterministic because I wanted it to be unique for all cases. Since I can figure out that resource group using tags, I think I can make users supply the rescue VM by ID or name. Thanks for the feedback.

Reply:
Some users may want to make multiple rescue VMs for the same faulty VM. For these cases, I need a way to create a unique resource group name which causes it to be non-deterministic. Is there any methods you have in mind to fix this problem?

Reply:
You can check for existence of a previous group and increment the suffix by one

`rescue-my-vm-1`, `rescue-my-vm-2`, etc.

Reply:
Then it wouldn't be deterministic at that point  right? Since it could be 1,2,3...

Reply:
Yes, it is is not deteministic with respect to the vm name, but if the user knows the index then the behavior of the command can be deterministic, in the case where a user wants to make multiple rescue vms per target vm.

given the target vm name and the "rescue vm id" (1,2,3...) you can figure out the resource group of the rescue vm. In this way, this id is actually a guid. In the case where this guid is not provided. You can default to the first resource group.

Reply:
This is just a suggestion though feel free to take this as you wish...